tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Configuration/RotationConfiguration.cs

```diff
@@ -18,60 +18,52 @@ public class RotationConfiguration
 
     public ReadOnlyCollection<PlanConfiguration> PlanConfigurations { get; }
 
-    public static RotationConfiguration From(string path,
+    public static RotationConfiguration From(IList<string> names,
+        IList<string> tags,
+        string directoryPath,
         IDictionary<string, Func<StoreConfiguration, SecretStore>> storeFactories)
     {
-        List<PlanConfiguration> planConfigurations = new();
+        IEnumerable<string> jsonFiles;
 
-        if (Directory.Exists(path))
+        try
         {
-            planConfigurations.AddRange(Directory.EnumerateFiles(path, "*.json", SearchOption.TopDirectoryOnly)
-                .Select(PlanConfiguration.FromFile));
+            jsonFiles = Directory
+                .EnumerateFiles(directoryPath, "*.json", SearchOption.AllDirectories);
         }
-        else
+        catch (Exception ex)
         {
-            planConfigurations.Add(PlanConfiguration.FromFile(path));
+            throw new RotationConfigurationException(
+                $"Error loading files in directory '{directoryPath}'.",
+                ex);
         }
 
-        var configuration = new RotationConfiguration(storeFactories, planConfigurations);
+        IEnumerable<PlanConfiguration> planConfigurations = jsonFiles
+            .Select(file => PlanConfiguration.TryLoadFromFile(file, out var plan) ? plan : null)
+            .Where(x => x != null)
+            .Select(x => x!);
 
-        return configuration;
-    }
-
-    public RotationPlan? GetRotationPlan(string name, ILogger logger, TimeProvider timeProvider)
-    {
-        PlanConfiguration? planConfiguration =
-            PlanConfigurations.FirstOrDefault(configuration => configuration.Name == name);
-
-        return planConfiguration != null
-            ? ResolveRotationPlan(planConfiguration, logger, timeProvider)
-            : null;
-    }
+        if (names.Any())
```

Comment:
I think if the name and tags arguments are passed, we should fail to the user as those are mutually exclusive arguments in my mind?

Reply:
Ah ok, I was thinking name was the actual file name, not a possibly ambiguous identifier.

Reply:
with folders names not being a component of the plan name, file name based plan names could still be ambiguous if the plans are in separate folders

Reply:
So will this scenario no longer work, or still work?

> secrets list --config ../path/to/config/allthingsinhere/

Reply:
Will work, won't apply any filtering to the plans found in the folder, recursive search